hrms/hr/doctype/appraisal/appraisal.py

```diff
@@ -179,9 +180,34 @@ def calculate_avg_feedback_score(self, update=False):
 			self.db_update()
 
 	def calculate_final_score(self):
-		final_score = (flt(self.total_score) + flt(self.avg_feedback_score) + flt(self.self_score)) / 3
-
-		self.final_score = flt(final_score, self.precision("final_score"))
+		if self.appraisal_cycle:
+			final_score = 0
+			appraisal_cycle_doc = frappe.get_doc("Appraisal Cycle", self.appraisal_cycle)
+			employee_doc = frappe.get_doc("Employee", self.employee)
+
+			formula = appraisal_cycle_doc.final_score_formula
+			based_on_formula = appraisal_cycle_doc.calculate_final_score_based_on_formula
```

Comment:
Move this inside the else block. We don't need this at all if its not based on formula. For the formula checkbox you can do a db call directly instead of get_doc

Reply:
resolved